switchbot/devices/lock.py

```diff
@@ -84,34 +84,34 @@ async def verify_encryption_key(
         return lock_info is not None
 
     @staticmethod
-    def api_request(subdomain: str, path: str, data: dict, headers: dict = None):
-        result = requests.post(
-            url=f"https://{subdomain}.{SWITCHBOT_APP_API_BASE_URL}/{path}",
-            headers=headers,
-            json=data,
-            timeout=5,
-        )
-
-        if result.status_code > 299:
-            raise SwitchbotApiError(
-                f"Unexpected status code returned by SwitchBot API: {result.status_code}"
-            )
-
-        response = json.loads(result.content)
-        if response["statusCode"] != 100:
-            raise SwitchbotApiError(
-                f"{response['message']}, status code: {response['statusCode']}"
-            )
-
-        return response["body"]
+    async def api_request(subdomain: str, path: str, data: dict = None, headers: dict = None):
```

Comment:
Please allow passing in an existing session (Home Assistant needs to do this so it doesn't make a new ssl context)